File: cpp/prime.cpp
#include <stdio.h> 
#include <cstdlib>


bool isPrime(int eingabe){
	for(int i = 2; i * i <= eingabe; i++){
		if(eingabe % i == 0){
			return false;
		}
	}
	return true;
}

int main(int argc, char* argv[]){

	int anzahl = atoi(argv[1]);
	
        int res, prev;
	for (int i = 2; i <= anzahl; i++) { 
		if(isPrime(i)){
                   res = i - prev;
		} else {
                   res = i - prev;
                }
                prev = res;
	}
        //printf("%i\n", res);
} 

File: csharp/prime.cs
public class prime {

    public static bool isPrime(int number) {
        for (int i = 2; i * i <= number; i++) {
            if (number % i == 0) {
                return false;
            }
        }
        return true;
    }
    
    
    public static int Main(string [] argv) {
        
        int limit = int.Parse(argv[0]);

        int res = 0;
        int prev = 0;
        for (int i = 2; i <= limit; i++) {
            if (isPrime(i)) {
               res = i - prev;
            } else {
               res = i - prev;
            }
            prev = res;
        }
        return 0;
    }
    
}
